app/app.go

```diff
@@ -484,172 +642,36 @@ func (app *App) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.Respo
-// store.
-//
-// Side-effect: calls baseapp.Init()
-func (app *App) Info(req abci.RequestInfo) abci.ResponseInfo {
-	if height := app.LastBlockHeight(); height > 0 {
-		ctx, err := app.CreateQueryContext(height, false)
-		if err != nil {
-			panic(err)
-		}
-		appVersion := app.GetAppVersionFromParamStore(ctx)
-		if appVersion > 0 {
-			app.SetAppVersion(ctx, appVersion)
-		} else {
-			app.SetAppVersion(ctx, v1)
-		}
-	}
-
-	resp := app.BaseApp.Info(req)
-	// mount the stores for the provided app version
-	if resp.AppVersion > 0 && !app.IsSealed() {
-		app.mountKeysAndInit(resp.AppVersion)
-	}
-
-	resp.Timeouts.TimeoutCommit = app.getTimeoutCommit(resp.AppVersion)
-	resp.Timeouts.TimeoutPropose = appconsts.GetTimeoutPropose(resp.AppVersion)
-
-	return resp
-}
-
-// InitChain implements the ABCI interface. This method is a wrapper around
-// baseapp's InitChain so that we can take the app version and setup the multicommit
-// store.
-//
-// Side-effect: calls baseapp.Init()
-func (app *App) InitChain(req abci.RequestInitChain) (res abci.ResponseInitChain) {
-	req = setDefaultAppVersion(req)
-	appVersion := req.ConsensusParams.Version.AppVersion
-	// mount the stores for the provided app version if it has not already been mounted
-	if app.AppVersion() == 0 && !app.IsSealed() {
-		app.mountKeysAndInit(appVersion)
-	}
-
-	res = app.BaseApp.InitChain(req)
-
-	ctx := app.NewContext(false, tmproto.Header{})
-	if appVersion != v1 {
-		app.SetInitialAppVersionInConsensusParams(ctx, appVersion)
-		app.SetAppVersion(ctx, appVersion)
-	}
-	res.Timeouts.TimeoutCommit = app.getTimeoutCommit(appVersion)
-	res.Timeouts.TimeoutPropose = appconsts.GetTimeoutPropose(appVersion)
-	return res
-}
-
-// setDefaultAppVersion sets the default app version in the consensus params if
-// it was 0. This is needed because chains (e.x. mocha-4) did not explicitly set
-// an app version in genesis.json.
-func setDefaultAppVersion(req abci.RequestInitChain) abci.RequestInitChain {
-	if req.ConsensusParams == nil {
-		panic("no consensus params set")
-	}
-	if req.ConsensusParams.Version == nil {
-		panic("no version set in consensus params")
-	}
-	if req.ConsensusParams.Version.AppVersion == 0 {
-		req.ConsensusParams.Version.AppVersion = v1
-	}
-	return req
-}
-
-// mountKeysAndInit mounts the keys for the provided app version and then
-// invokes baseapp.Init().
-func (app *App) mountKeysAndInit(appVersion uint64) {
-	app.Logger().Info(fmt.Sprintf("mounting KV stores for app version %v", appVersion))
-	app.MountKVStores(app.versionedKeys(appVersion))
-
-	// Invoke load latest version for its side-effect of invoking baseapp.Init()
-	if err := app.LoadLatestVersion(); err != nil {
-		panic(fmt.Sprintf("loading latest version: %s", err.Error()))
-	}
+	// REVIEW: these are moved to YAML, ok to delete?
+	// res.Timeouts.TimeoutCommit = app.getTimeoutCommit(currentVersion)
+	// res.Timeouts.TimeoutPropose = appconsts.GetTimeoutPropose(currentVersion)
+	return res, nil
```

Comment:
setting them here allows for the network to change all at once during a rolling upgrade. iirc, someone mentioned comet might have support for this natively in v1, but I was unsure.

no matter what, we need to the timeouts to change when the app version changes. if we don't need to modify the sdk or comet to do that, then I think